src/main/resources/scripts/cleanup/non_prod_primary_and_replica_data_cleanup.sql

```diff
@@ -0,0 +1,92 @@
+--
+-- RWA-2482 cleanup old tasks from primary task database
+--
+
+CREATE OR REPLACE FUNCTION cft_task_db.task_cleanup_between_dates_primary()
+RETURNS VOID
+LANGUAGE plpgsql
+AS $function$
+DECLARE
+    task_id_var text;
+BEGIN
+    -- Loop through the rows of the tasks table
+    FOR task_id_var IN SELECT task_id FROM cft_task_db.tasks WHERE created >= '<From Date and time>' AND created <= '<To Date and time>' limit 10000 LOOP
+
+        -- delete from sensitive_task_event_logs
+        DELETE FROM cft_task_db.sensitive_task_event_logs WHERE task_id = task_id_var;
+
+        IF EXISTS (SELECT 1 FROM cft_task_db.sensitive_task_event_logs WHERE task_id = task_id_var) THEN
+            RAISE NOTICE 'Issue with the taskId: % in sensitive_task_event_logs table. Skipping next steps', task_id_var;
+            CONTINUE;
```

Comment:
If there is no sensitive log entry it should not stop the reset of the deletion.  Can be zero or many sensitive logs per task_id.